src/drt/src/dr/AbstractDRGraphics.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "frBaseTypes.h"
+
+namespace gui {
+class Painter;
+}
+
+namespace odb {
+class dbDatabase;
+class Point;
+class dbTechLayer;
+}  // namespace odb
+
+namespace drt {
+
+class FlexGridGraph;
+class FlexWavefrontGrid;
+class FlexDRWorker;
+class drNet;
+class frDesign;
+class frBlockObject;
+struct RouterConfiguration;
+
+class AbstractDRGraphics
```

Comment:
I believe many of these methods don't need to be defined in this interface.  Only methods called from the rest of dr need to be here.  For example status() and methods from Renderer API.  Likewise with the other abstract base classes for graphics.